src/main/java/org/opendatakit/suitcase/ui/IOPanel.java

```diff
@@ -10,25 +10,35 @@
   public static final String PUSH_TAB_LABEL = "Upload";
 
   private MainPanel parent;
+  private PullPanel pullPanel;
+  private PushPanel pushPanel;
 
   public IOPanel(MainPanel parent) {
     super(new GridBagLayout());
 
     this.parent = parent;
 
     JTabbedPane tabs = new JTabbedPane();
-    tabs.addTab(PULL_TAB_LABEL, new PullPanel(this));
-    tabs.addTab(PUSH_TAB_LABEL, new PushPanel(this));
+    pullPanel = new PullPanel(this);
+    pushPanel = new PushPanel(this);
+    tabs.addTab(PULL_TAB_LABEL, pullPanel);
+    tabs.addTab(PUSH_TAB_LABEL, pushPanel);
 
     JSplitPane splitPane =
-        new JSplitPane(JSplitPane.VERTICAL_SPLIT, tabs, parent.getProgressBar());
+            new JSplitPane(JSplitPane.VERTICAL_SPLIT, tabs, parent.getProgressBar());
     splitPane.setResizeWeight(0.8);
     splitPane.setDividerSize(0);
     splitPane.setEnabled(false);
 
     this.add(splitPane, LayoutDefault.getDefaultGbc());
   }
 
+  public void setButtonState(boolean state)
```

Comment:
This function name is confusing because there are multiple buttons and if you only look at the function name and parameters it is not clear what will happen. 

I realize there was another private function that did that, but being more descriptive makes reading code easier and understanding the intent of what the function is supposed to do.